lib/api_auth/request_drivers/faraday.rb

```diff
@@ -56,7 +56,9 @@ def content_md5
       end
 
       def request_uri
-        uri = URI::HTTP.new(nil, nil, nil, nil, nil, @request.path, nil, @request.params.to_query, nil)
+        query_string = @request.params.map{|k,v| "#{k}=#{v}"}.join('&')
```

Comment:
You are correct! (I did a quick test too).
Thanks, that will make it a lot better.